lib/Github/Api/Repo.php

```diff
@@ -150,6 +150,23 @@ public function show($username, $repository)
     {
         return $this->get('/repos/'.rawurlencode($username).'/'.rawurlencode($repository));
     }
+    
+    /**
+     * Get extended information about a repository by its id.
+     * Note: at time of writing this is an undocumented feature but GitHub support have advised that it can be relied on.
+     *
+     * @link http://developer.github.com/v3/repos/
+     * @link https://github.com/piotrmurach/github/issues/283
+     * @link https://github.com/piotrmurach/github/issues/282
+     *
+     * @param int $id   the id of the repository
+     *
+     * @return array information about the repository
+     */
+    public function showById($id)
+    {
+        return $this->get('/repositories/'.rawurlencode($id));
```

Comment:
Yes, it should always be an integer (currently anyway). I thought about checking that they'd input an integer and if not throwing an exception, but then figured that it was better than us assuming IDs never become non-integers to just pass it over to GitHub and let them respond accordingly. Having the `rawurlencode` ensures that whatever is passed into the function (even if it shouldn't have been) is sent over to GitHub properly.

There probably isn't much in it as I think it's highly unlikely that GH will ever start to issue non-integer IDs, but personally I prefer this way - it's GitHub's ID, let's let them validate it.